alphagov/paas-admin
# [#156916095] Generating monthly usage reports

## What

We have moved the basic UI for rendering usage reports from paas-billing to paas-admin so that we can use it to generate bills. To do this we implemented a BillingClient to talk to our billing API, added a new handler for fetching/grouping the events for an org by "name" and sketched out rendering the data table.

* **This is NOT the final design of the report**: a future story will implement the new designs that have been created, it is intended for internal use for now.
* **This is NOT currently exposed to all tenants**: there is no "billing" link in the UI (you must know the url to visit it) and paas-billing currently restricts access to tokens with `admin`, `admin_read_only` or `global_auditor` scope. A future story will expose this feature for all tenants with BillingManager and/or OrgManager roles.

## How to review

* Code review
* Run the application against a dev environment that has some billing-db data.
* Visit `/organisations/{org_guid}/statements/{YYYY-MM-DD}`
* Check that things make sense

## Who can review

Not @chrisfarms or @paroxp
